### PR TITLE
feat(birdie): file tree view for asset explorer

### PR DESCRIPTION
**Pull request by sylviacx**
7298923805

### CODE REVIEW

**Comment by sylviacx** [comment]
minor visual bug, switching from file tree -> table asset explorer, first time opening directories will result in lock icons missing, but if you reopen the directory it will show up correctly

File: birdie/src/lib/components/files/TreeNode.svelte [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+		fileNode = {
+			...fileNode, // spread the existing fileNode
+			children: newChildren
+		};
+	};
+
+	const handleOnClick = () => {
+		loading = true;
+		console.log('loading', loading);
```
Comment (sylviacx) [inline birdie/src/lib/components/files/TreeNode.svelte:42]: remove console.logs

File: birdie/src/lib/components/files/TreeNode.svelte [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	};
+
+	const handleOnClick = () => {
+		loading = true;
+		fileNode = {
+			...fileNode,
+			open: !fileNode.open
+		};
+		if (fileNode.open) {
+			void getChildren();
```
Comment (JulianRushdown) [inline birdie/src/lib/components/files/TreeNode.svelte:47]: curious should this getChildren call be awaited here so this handler doesn't accidentally set `loading = false` before the getChildren call finishes up?

Comment (sylviacx) [reply birdie/src/lib/components/files/TreeNode.svelte:47]: good catch, yea this whole function should be async

File: birdie/src-tauri/src/repo/file.rs [hunk 3]
```diff
@@ -42,10 +43,98 @@ pub struct File {
...
+            }
+            FileType::Directory => LocalFileLFSState::None,
+        };
+
+        let size = match lfs_state {
+            LocalFileLFSState::Stub => {
+                let reader = BufReader::new(fs::File::open(file_path)?);
+
+                // get 3rd line, split by space, get 2nd element, parse to u64
+                let size: u64 = reader
```
Comment (JulianRushdown) [inline birdie/src-tauri/src/repo/file.rs:100]: Not sure if Tauri gracefully handles the panics that might come from the unwraps here, but chaining `and_thens` might make it slightly easier to read. maybe something like:

`
reader
            .lines()
            .nth(2)
            .and_then(|line| line.ok())
            .and_then(|line| line.split(' ').nth(1))
            .and_then(|size_str| size_str.parse().ok())
            .ok_or_else(|| Err(create_tauri_error(res).await))?
`

**Review by rudoi — APPROVED** [review]
yaaaaaaa

**Pull request merged**